paymentsheet/src/main/java/com/stripe/android/link/ui/wallet/WalletViewModel.kt

```diff
@@ -192,6 +199,45 @@ internal class WalletViewModel @Inject constructor(
         )
     }
 
+    private fun shouldAutoSelectDefaultPaymentMethod(): Boolean {
+        return linkLaunchMode is LinkLaunchMode.PaymentMethodSelection &&
+            configuration.skipWalletInFlowController
+    }
+
+    private suspend fun handleAutoSelection(paymentDetails: List<LinkPaymentMethod.ConsumerPaymentDetails>) {
+        val autoSelectedPaymentMethod =
+            paymentDetails.singleOrNull()?.details
+                ?: paymentDetails.firstOrNull { it.details.isDefault }?.details
+        _uiState.update { it.copy(hasAttemptedAutoSelection = true) }
+
+        if (autoSelectedPaymentMethod?.isReadyForUse() == true) {
+            // Set the default as selected and proceed with payment selection
+            _uiState.update {
+                it.copy(selectedItemId = autoSelectedPaymentMethod.id)
+            }
+            performPaymentConfirmation(autoSelectedPaymentMethod)
+        } else {
+            // Auto-selection not supported, show the wallet UI
+            _uiState.update {
+                it.copy(isAutoSelecting = false)
+            }
+        }
+    }
+
+    private fun ConsumerPaymentDetails.PaymentDetails.isReadyForUse(): Boolean {
+        // Check if card requires details recollection (includes both expiry and CVC checks)
+        val requiresCardDetailsRecollection = (this as? ConsumerPaymentDetails.Card)
+            ?.requiresCardDetailsRecollection == true
+
+        // Check if billing details collection is needed
+        val needsBillingDetails = supports(
+            billingDetailsConfig = configuration.billingDetailsCollectionConfiguration,
+            linkAccount = linkAccount
+        ).not() && _uiState.value.collectMissingBillingDetailsForExistingPaymentMethods
```

Comment:
Should we also attempt to prefill fields from the Link account?

Reply:
Chatted about this offline - we'll be keeping the existing behavior and let the user confirm the merged billing details on Android.